# Flight look-to-book Data Preprocessing

#### @author glin
#### 15/06/2019

In [1]:
import pandas as pd

**Question 1**

Check the number of lines in each of the two files (bookings and searches)

In [7]:
bookingPath_small = '../data/sample/bookings.csv.small'
bookings_small = pd.read_csv(bookingPath_small, header=0, sep='^')

In [8]:
bookings_small

act_date             source  pos_ctry                          pos_iata  \
0  2013-03-05 00:00:00  1A      DE        a68dd7ae953c8acfb187a1af2dcbe123   
1  2013-03-26 00:00:00  1A      US        e612b9eeeee6f17f42d9b0d3b79e75ca   
2  2013-03-26 00:00:00  1A      US        e612b9eeeee6f17f42d9b0d3b79e75ca   
3  2013-03-26 00:00:00  1A      AU        0f984b3bb6bd06661c95529bbd6193bc   
4  2013-03-26 00:00:00  1A      AU        0f984b3bb6bd06661c95529bbd6193bc   
5  2013-03-20 00:00:00  1V      US        0283592e45e392871f7e14b1e22a9aaf   
6  2013-03-20 00:00:00  1V      US        0283592e45e392871f7e14b1e22a9aaf   
7  2013-03-25 00:00:00  1V      JP        5af045902bd23cab579915611d99e1e0   
8  2013-03-25 00:00:00  1V      JP        5af045902bd23cab579915611d99e1e0   
9  2013-03-25 00:00:00  1V      JP        5af045902bd23cab579915611d99e1e0   

                          pos_oid                      rloc            \
0  1a11ae49fcbf545fd2afc1a24d88d2b7  ea65900e72d71f4626378e2ebd298267   
1  7437560d8f276d6d05eeb806d9e7edee  737295a86982c941f1c2da9a46a14043   
2  7437560d8f276d6d05eeb806d9e7edee  737295a86982c941f1c2da9a46a14043   
3  36472c6dbaf7afec9136ac40364e2794  5ecf00fdcbcec761c43dc7285253d0c1   
4  36472c6dbaf7afec9136ac40364e2794  5ecf00fdcbcec761c43dc7285253d0c1   
5  2d0f8e7e1bb792625d2f34f221bc38f1  d54708298686de200f8270e7c637af5e   
6  2d0f8e7e1bb792625d2f34f221bc38f1  d54708298686de200f8270e7c637af5e   
7  5073861d8597467c33596bfe16f23c56  a37584d1485cb35991e4ff1a2ba92262   
8  5073861d8597467c33596bfe16f23c56  a37584d1485cb35991e4ff1a2ba92262   
9  5073861d8597467c33596bfe16f23c56  a37584d1485cb35991e4ff1a2ba92262   

   cre_date             duration  distance  dep_port  ...  route            \
0  2013-02-22 00:00:00      1708         0  ZRH       ...  LHRZRH            
1  2013-03-26 00:00:00    135270         0  SAL       ...  SALATLCLT         
2  2013-03-26 00:00:00    135270         0  SAL       ...  CLTATLSAL         
3  2013-03-26 00:00:00     30885         0  AKL       ...  AKLHKGSVO         
4  2013-03-26 00:00:00     30885         0  AKL       ...  SVOHKGAKL         
5  2013-03-20 00:00:00      5923         0  DEN       ...  DENLGA            
6  2013-03-20 00:00:00      5923         0  DEN       ...  LGADEN            
7  2013-03-25 00:00:00      8371        60  NRT       ...  NRTSIN            
8  2013-03-25 00:00:00      8371        60  NRT       ...  SINPEN            
9  2013-03-25 00:00:00      8371        60  NRT       ...  PENBKK            

  carrier  bkg_class  cab_class  brd_time             off_time            pax  \
0      VI  T          Y          2013-03-07 08:50:00  2013-03-07 11:33:37  -1   
1      NV  L          Y          2013-04-12 13:04:00  2013-04-12 22:05:40   1   
2      NV  U          Y          2013-07-15 07:00:00  2013-07-15 11:34:51   1   
3      XK  G          Y          2013-04-24 23:59:00  2013-04-25 16:06:31   1   
4      XK  G          Y          2013-05-14 20:15:00  2013-05-16 10:44:50   1   
5      FK  W          Y          2013-04-07 10:23:00  2013-04-07 16:14:24   1   
6      FK  K          Y          2013-04-11 11:15:00  2013-04-11 13:06:24   1   
7      XR  Q          Y          2013-04-14 11:05:00  2013-04-14 17:10:56   2   
8      WS  Y          Y          2013-04-16 15:45:00  2013-04-16 17:15:29   2   
9      EQ  Y          Y          2013-04-18 08:00:00  2013-04-18 08:53:56   2   

   year month  oid        
0  2013     3  NULL       
1  2013     3  NULL       
2  2013     3  NULL       
3  2013     3  SYDA82546  
4  2013     3  SYDA82546  
5  2013     3  NULL       
6  2013     3  NULL       
7  2013     3  NULL       
8  2013     3  NULL       
9  2013     3  NULL       

[10 rows x 38 columns]

In [9]:
bookings_small.columns

Index(['act_date           ', 'source', 'pos_ctry', 'pos_iata', 'pos_oid  ',
       'rloc          ', 'cre_date           ', 'duration', 'distance',
       'dep_port', 'dep_city', 'dep_ctry', 'arr_port', 'arr_city', 'arr_ctry',
       'lst_port', 'lst_city', 'lst_ctry', 'brd_port', 'brd_city', 'brd_ctry',
       'off_port', 'off_city', 'off_ctry', 'mkt_port', 'mkt_city', 'mkt_ctry',
       'intl', 'route          ', 'carrier', 'bkg_class', 'cab_class',
       'brd_time           ', 'off_time           ', 'pax', 'year', 'month',
       'oid      '],
      dtype='object')

In [10]:
len(bookings_small.index)

10

In [15]:
fields = ['arr_port','pax']
%time bookings = pd.read_csv(bookingPath, header=0, sep='^', usecols=fields)
len(bookings.index)

CPU times: user 19.8 s, sys: 1.39 s, total: 21.2 s
Wall time: 27 s


10000010

In [16]:
searchesPath = '../../searches.csv'
fields = ['Destination', 'Date']
%time searches = pd.read_csv(searchesPath, header=0, sep='^', usecols=fields)
len(searches.index)

CPU times: user 23 s, sys: 1.48 s, total: 24.4 s
Wall time: 30 s


20390198

**Solution to Question 1**
Number of lines of bookings file: 10000010
Number of lines of searches file: 20390198

########################################################

**Question 2**

The arrival airport is the column arr_port. It is the IATA code for the airport.

To get the total number of passengers for an airport, you can sum the column pax, grouping by arr_port. Note that there is negative pax. That corresponds to cancelations. So to get the total number of passengers that have actually booked, you should sum including the negatives (that will remove the canceled bookings).

Print the top 10 arrival airports in the standard output, including the number of passengers.

Required: Get the name of the city or airport corresponding to that airport (programmatically, we suggest to have a look at GeoBases in Github)

If you’re using Python, can you solve this problem using pandas?

In [17]:
bookings.columns = bookings.columns.str.strip()

In [ ]:
preGrouped = bookings.groupby(['arr_port'])['pax'].sum()

In [19]:
type(preGrouped)

pandas.core.series.Series

In [21]:
grouped = pd.DataFrame({'arr_port':preGrouped.index, 'acc_pax':preGrouped.values})

In [22]:
grouped

arr_port  acc_pax
0     AAB          30.0
1     AAE         810.0
2     AAL         810.0
3     AAN          20.0
4     AAQ        1650.0
5     AAR         500.0
6     ABA          90.0
7     ABB          40.0
8     ABE         880.0
9     ABI         150.0
10    ABJ        2600.0
11    ABK           0.0
12    ABQ        4290.0
13    ABR           0.0
14    ABS         310.0
15    ABT         300.0
16    ABV        1950.0
17    ABX         300.0
18    ABY          90.0
19    ABZ        3710.0
20    ACA         700.0
21    ACC        3810.0
22    ACE        1700.0
23    ACH          20.0
24    ACI          40.0
25    ACK         200.0
26    ACT         180.0
27    ACV         140.0
28    ACY         190.0
29    ADA         980.0
...        ...      ...
2245  ZAZ          50.0
2246  ZBF         110.0
2247  ZBL          90.0
2248  ZCL         150.0
2249  ZCO         490.0
2250  ZDH          60.0
2251  ZFQ          40.0
2252  ZFV         120.0
2253  ZHA           0.0
2254  ZIG          40.0
2255  ZIH         870.0
2256  ZLN           0.0
2257  ZLO         500.0
2258  ZNE        1510.0
2259  ZNZ         970.0
2260  ZOS          70.0
2261  ZQN        1720.0
2262  ZRH       20190.0
2263  ZSA          10.0
2264  ZSE          10.0
2265  ZTH          50.0
2266  ZUH          40.0
2267  ZVE          40.0
2268  ZVJ          20.0
2269  ZVK          20.0
2270  ZWE          30.0
2271  ZWS         620.0
2272  ZYL        1290.0
2273  ZYN          20.0
2274  ZYR        1020.0

[2275 rows x 2 columns]

In [24]:
geo = pd.read_csv('../data/geobase/airports_geonames_only_clean.csv', sep='^', header=None).iloc[:,0:2]

In [25]:
geo

0                                     1
0     AUH       Abu Dhabi International Airport
1     AZI              Abu Dhabi Bateen Airport
2     AAN          Al Ain International Airport
3     DXB           Dubai International Airport
4     FJR                              Fujairah
5     RKT  Ras Al Khaimah International Airport
6     SHJ         Sharjah International Airport
7     KBL                   Kabul International
8     FAH                                 Farah
9     FBD                              Faizabad
10    HEA                                 Herat
11    JAA                             Jalalabad
12    MZR                        Mazar-I-Sharif
13    ANU      V. C. Bird International Airport
14    AXA                    Wall Blake Airport
15    TIA                          Tirane/Rinas
16    EVN                             Zvartnots
17    LWN                                Shirak
18    SXM              Princess Juliana Airport
19    EUX     Franklin Delano Roosevelt Airport
20    BON                      Flamingo Airport
21    CUR         Curacao International Airport
22    BUG                     Monbaca Benguela 
23    CAB                               Cabinda
24    NOV                   Huambo Nova Lisboa 
25    SVP                      Bie Silva Porto 
26    LAD                                Luanda
27    MDQ                         Mar Del Plata
28    ROS                 Rosario International
29    COC                   Concordia Aerodrome
...   ...                                   ...
3529  CIP                               Chipata
3530  KMZ                                 Kaoma
3531  KAA                                Kasama
3532  QKE                                 Kabwe
3533  LVI                           Livingstone
3534  LUN           Lusaka Internationalairport
3535  MNS                                 Mansa
3536  MFU                                 Mfuwe
3537  MNR                                 Mongu
3538  NLA                                 Ndola
3539  SXG                               Senanga
3540  SJQ                               Sesheke
3541  SLI                               Solwezi
3542  BBZ                               Zambezi
3543  KIW                            Southdowns
3544  BUQ                      Bulawayo Airport
3545  HRE          Harare International Airport
3546  CHJ                              Chipinge
3547  BFO                         Buffalo Range
3548  VFA                        Victoria Falls
3549  KAB                                Kariba
3550  MVZ                              Masvingo
3551  GWE                                 Gweru
3552  HWN                  Hwange National Park
3553  MPN                Mount Pleasant Airport
3554  OHD                                 Ohrid
3555  SKP                       Skopje-Petrovec
3556  PSY                       Stanley Airport
3557  TIH                         Tikehau Atoll
3558  MOZ                                 Temae

[3559 rows x 2 columns]

In [26]:
geo.columns = ['arr_port', 'airport']

In [27]:
merged = pd.merge(grouped, geo, on=['arr_port'])

In [28]:
merged

Empty DataFrame
Columns: [arr_port, acc_pax, airport]
Index: []

In [29]:
grouped

arr_port  acc_pax
0     AAB          30.0
1     AAE         810.0
2     AAL         810.0
3     AAN          20.0
4     AAQ        1650.0
5     AAR         500.0
6     ABA          90.0
7     ABB          40.0
8     ABE         880.0
9     ABI         150.0
10    ABJ        2600.0
11    ABK           0.0
12    ABQ        4290.0
13    ABR           0.0
14    ABS         310.0
15    ABT         300.0
16    ABV        1950.0
17    ABX         300.0
18    ABY          90.0
19    ABZ        3710.0
20    ACA         700.0
21    ACC        3810.0
22    ACE        1700.0
23    ACH          20.0
24    ACI          40.0
25    ACK         200.0
26    ACT         180.0
27    ACV         140.0
28    ACY         190.0
29    ADA         980.0
...        ...      ...
2245  ZAZ          50.0
2246  ZBF         110.0
2247  ZBL          90.0
2248  ZCL         150.0
2249  ZCO         490.0
2250  ZDH          60.0
2251  ZFQ          40.0
2252  ZFV         120.0
2253  ZHA           0.0
2254  ZIG          40.0
2255  ZIH         870.0
2256  ZLN           0.0
2257  ZLO         500.0
2258  ZNE        1510.0
2259  ZNZ         970.0
2260  ZOS          70.0
2261  ZQN        1720.0
2262  ZRH       20190.0
2263  ZSA          10.0
2264  ZSE          10.0
2265  ZTH          50.0
2266  ZUH          40.0
2267  ZVE          40.0
2268  ZVJ          20.0
2269  ZVK          20.0
2270  ZWE          30.0
2271  ZWS         620.0
2272  ZYL        1290.0
2273  ZYN          20.0
2274  ZYR        1020.0

[2275 rows x 2 columns]

In [30]:
geo

arr_port                               airport
0         AUH       Abu Dhabi International Airport
1         AZI              Abu Dhabi Bateen Airport
2         AAN          Al Ain International Airport
3         DXB           Dubai International Airport
4         FJR                              Fujairah
5         RKT  Ras Al Khaimah International Airport
6         SHJ         Sharjah International Airport
7         KBL                   Kabul International
8         FAH                                 Farah
9         FBD                              Faizabad
10        HEA                                 Herat
11        JAA                             Jalalabad
12        MZR                        Mazar-I-Sharif
13        ANU      V. C. Bird International Airport
14        AXA                    Wall Blake Airport
15        TIA                          Tirane/Rinas
16        EVN                             Zvartnots
17        LWN                                Shirak
18        SXM              Princess Juliana Airport
19        EUX     Franklin Delano Roosevelt Airport
20        BON                      Flamingo Airport
21        CUR         Curacao International Airport
22        BUG                     Monbaca Benguela 
23        CAB                               Cabinda
24        NOV                   Huambo Nova Lisboa 
25        SVP                      Bie Silva Porto 
26        LAD                                Luanda
27        MDQ                         Mar Del Plata
28        ROS                 Rosario International
29        COC                   Concordia Aerodrome
...       ...                                   ...
3529      CIP                               Chipata
3530      KMZ                                 Kaoma
3531      KAA                                Kasama
3532      QKE                                 Kabwe
3533      LVI                           Livingstone
3534      LUN           Lusaka Internationalairport
3535      MNS                                 Mansa
3536      MFU                                 Mfuwe
3537      MNR                                 Mongu
3538      NLA                                 Ndola
3539      SXG                               Senanga
3540      SJQ                               Sesheke
3541      SLI                               Solwezi
3542      BBZ                               Zambezi
3543      KIW                            Southdowns
3544      BUQ                      Bulawayo Airport
3545      HRE          Harare International Airport
3546      CHJ                              Chipinge
3547      BFO                         Buffalo Range
3548      VFA                        Victoria Falls
3549      KAB                                Kariba
3550      MVZ                              Masvingo
3551      GWE                                 Gweru
3552      HWN                  Hwange National Park
3553      MPN                Mount Pleasant Airport
3554      OHD                                 Ohrid
3555      SKP                       Skopje-Petrovec
3556      PSY                       Stanley Airport
3557      TIH                         Tikehau Atoll
3558      MOZ                                 Temae

[3559 rows x 2 columns]

In [31]:
merged = pd.merge(grouped, geo, on=['arr_port'])
merged

Empty DataFrame
Columns: [arr_port, acc_pax, airport]
Index: []

In [32]:
grouped['arr_port'] = grouped['arr_port'].str.strip()

In [33]:
merged = pd.merge(grouped, geo, on=['arr_port'])
merged

arr_port  acc_pax                                            airport
0         AAE    810.0                                             Annaba
1         AAL    810.0                                             Alborg
2         AAN     20.0                       Al Ain International Airport
3         AAQ   1650.0                                  Anapa / Vitiazevo
4         AAR    500.0                                           Tirstrup
5         ABA     90.0                                             Abakan
6         ABE    880.0                                     Martin Airport
7         ABI    150.0                           Abilene Regional Airport
8         ABJ   2600.0                                            Abidjan
9         ABK      0.0                               Kabre Dare Military 
10        ABQ   4290.0                  Albuquerque International Airport
11        ABR      0.0                          Aberdeen Regional Airport
12        ABS    310.0                                         Abu Simbel
13        ABT    300.0                                            Al Baha
14        ABX    300.0                                     Albury Airport
15        ABY     90.0                 Southwest Georgia Regional Airport
16        ABZ   3710.0                                    Aberdeen / Dyce
17        ACA    700.0                              Acapulco / G. Alvarez
18        ACC   3810.0                       Kotoka International Airport
19        ACE   1700.0                             Lanzarote / Aeropuerto
20        ACH     20.0                            Saint Gallen-Altenrhein
21        ACI     40.0                          Alderney / Channel Island
22        ACK    200.0                         Nantucket Memorial Airport
23        ACT    180.0                              Waco Regional Airport
24        ACV    140.0                                     Arcata Airport
25        ACY    190.0                Atlantic City International Airport
26        ADA    980.0                                  Adana / Sakirpasa
27        ADB   3060.0                             Izmir / Adnan Menderes
28        ADD   5880.0                                        Addis Ababa
29        ADE    600.0                                               Aden
...       ...      ...                                                ...
1665      YNT    430.0                                            Laishan
1666      YOW   3860.0                                 Ottawa Int'L. Ont.
1667      YQB   1580.0                  Jean Lesage International Airport
1668      YQR   1400.0                                             Regina
1669      YUL  14040.0  Montreal / Pierre Elliot Trudeau International...
1670      YUM    410.0                         Yuma International Airport
1671      YVR  15920.0            Vancouver International Airport;  B. C.
1672      YWG   2680.0                          Winnipeg Int. Airportman.
1673      YXH     30.0                               Medicine Hat;  Alta.
1674      YXU    450.0                                      London;  Ont.
1675      YYC   9810.0                      Calgary International;  Alta.
1676      YYJ   1290.0                      Victoria Int. Airport;  B. C.
1677      YYZ  26070.0                        Toronto Pearson Int'L. Ont.
1678      ZAD     90.0                                              Zadar
1679      ZAG   3610.0                                     Zagreb / Pleso
1680      ZAL     90.0                                           Valdivia
1681      ZAM     80.0                                          Zamboanga
1682      ZCL    150.0                                  Zacatecas Airport
1683      ZCO    490.0                                             Temuco
1684      ZHA      0.0                                          Zhanjiang
1685      ZIG     40.0                                         Ziguinchor
1686      ZIH    870.0                                 Ixtapa-Zihuatanejo
1687     

In [34]:
merged.sort_values(by=['acc_pax'], ascending=False).head(10)

arr_port  acc_pax                               airport
883       LHR  88809.0               London Heathrow Airport
960       MCO  70930.0         Orlando International Airport
853       LAX  70530.0     Los Angeles International Airport
851       LAS  69630.0        McCarran International Airport
728       JFK  66270.0  John F Kennedy International Airport
272       CDG  64490.0             Paris - Charles-de-Gaulle
190       BKK  59460.0                          Suvarnabhumi
990       MIA  58150.0           Miami International Airport
1357      SFO  58000.0   San Francisco International Airport
438       DXB  55590.0           Dubai International Airport

**Solution to question 2**
See above

########################################################

**Question 3**

Plot the monthly number of searches for flights arriving at Malaga, Madrid or Barcelona

For the arriving airport, you can use the Destination column in the searches file. Plot a curve for Malaga, another one for Madrid, and another one for Barcelona, in the same figure.

If you’re using Python, can you solve this problem using pandas?

In [35]:
sg = pd.merge(searches, geo, how='left', left_on=['Destination'], right_on=['arr_port'])

In [36]:
sg

Date Destination arr_port  \
0         2013-01-01         AUH      AUH   
1         2013-01-01         MIL      NaN   
2         2013-01-01         SFO      SFO   
3         2013-01-01         ARN      ARN   
4         2013-01-01         MAD      MAD   
5         2013-01-01         BLR      BLR   
6         2013-01-01         PAR      NaN   
7         2013-01-01         DUB      DUB   
8         2013-01-01         ACE      ACE   
9         2013-01-01         BGW      NaN   
10        2013-01-01         BUD      BUD   
11        2013-01-01         POA      POA   
12        2013-01-01         MXP      MXP   
13        2013-01-01         LON      NaN   
14        2013-01-01         SNN      SNN   
15        2013-01-01         SLZ      SLZ   
16        2013-01-01         PHL      PHL   
17        2013-01-01         GRU      GRU   
18        2013-01-01         IKT      IKT   
19        2013-01-01         HKG      HKG   
20        2013-01-01         SEA      SEA   
21        2013-01-01         STN      STN   
22        2013-01-01         FUE      FUE   
23        2013-01-01         IST      IST   
24        2013-01-01         VIE      VIE   
25        2013-01-01         SAO      SAO   
26        2013-01-01         BRI      BRI   
27        2013-01-01         JED      JED   
28        2013-01-01         CPH      CPH   
29        2013-01-01         DRS      DRS   
...              ...         ...      ...   
20390168  2013-10-13         ZAD      ZAD   
20390169  2013-10-13         PEI      PEI   
20390170  2013-10-13         MLE      MLE   
20390171  2013-10-13         DEL      DEL   
20390172  2013-10-13         ABJ      ABJ   
20390173  2013-10-13         BRU      BRU   
20390174  2013-10-13         CDG      CDG   
20390175  2013-10-13         BKK      BKK   
20390176  2013-10-13         SSA      SSA   
20390177  2013-10-13         CPH      CPH   
20390178  2013-10-13         CGH      CGH   
20390179  2013-10-13         PVR      PVR   
20390180  2013-10-13         HAM      HAM   
20390181  2013-10-13         PRG      PRG   
20390182  2013-10-13         JFK      JFK   
20390183  2013-10-13         ICN      ICN   
20390184  2013-10-13         PMI      PMI   
20390185  2013-10-13         BKK      BKK   
20390186  2013-10-13         AMS      AMS   
20390187  2013-10-13         SRP      SRP   
20390188  2013-10-13         MCT      MCT   
20390189  2013-10-13         SKG      SKG   
20390190  2013-10-13         AMD      AMD   
20390191  2013-10-13         BRS      BRS   
20390192  2013-10-13         AKL      AKL   
20390193  2013-10-13         NBO      NBO   
20390194  2013-10-13         ALA      ALA   
20390195  2013-10-13         LHR      LHR   
20390196  2013-10-13         MAD      MAD   
20390197  2013-10-13          HA      NaN   

                                                    airport  
0                           Abu Dhabi International Airport  
1                                                       NaN  
2                       San Francisco International Airport  
3                                                   Arlanda  
4                                          Madrid / Barajas  
5                                     Bangalore / Hindustan  
6                                                       NaN  
7                                            Dublin Airport  
8                                    Lanzarote / Aeropuerto  
9                                                       NaN  
10                                      Budapest / Ferihegy  
11                    Aeroporto Internacional Salgado Filho  
12                                 Aeroporto della Malpensa  
13                                                      NaN  
14                                          Shannon Airport  
15                                      Sao Luiz Aeroporto   
16                       Philadelphia International Airport  
17                                        Guarulhos Airport  
18                                             

In [38]:
df = sg.loc[sg['airport'].str.contains('Madrid')==True]

In [39]:
df

Date Destination arr_port           airport
4         2013-01-01         MAD      MAD  Madrid / Barajas
151       2013-01-01         MAD      MAD  Madrid / Barajas
375       2013-01-01         MAD      MAD  Madrid / Barajas
406       2013-01-01         MAD      MAD  Madrid / Barajas
509       2013-01-01         MAD      MAD  Madrid / Barajas
606       2013-01-01         MAD      MAD  Madrid / Barajas
688       2013-01-01         MAD      MAD  Madrid / Barajas
791       2013-01-01         MAD      MAD  Madrid / Barajas
798       2013-01-01         MAD      MAD  Madrid / Barajas
895       2013-01-01         MAD      MAD  Madrid / Barajas
897       2013-01-01         MAD      MAD  Madrid / Barajas
1026      2013-01-02         MAD      MAD  Madrid / Barajas
1038      2013-01-02         MAD      MAD  Madrid / Barajas
1061      2013-01-02         MAD      MAD  Madrid / Barajas
1117      2013-01-02         MAD      MAD  Madrid / Barajas
1233      2013-01-02         MAD      MAD  Madrid / Barajas
1407      2013-01-02         MAD      MAD  Madrid / Barajas
1584      2013-01-02         MAD      MAD  Madrid / Barajas
1619      2013-01-02         MAD      MAD  Madrid / Barajas
1700      2013-01-02         MAD      MAD  Madrid / Barajas
1715      2013-01-02         MAD      MAD  Madrid / Barajas
1790      2013-01-02         MAD      MAD  Madrid / Barajas
2162      2013-01-03         MAD      MAD  Madrid / Barajas
2280      2013-01-03         MAD      MAD  Madrid / Barajas
2813      2013-01-03         MAD      MAD  Madrid / Barajas
2856      2013-01-03         MAD      MAD  Madrid / Barajas
2963      2013-01-03         MAD      MAD  Madrid / Barajas
3077      2013-01-04         MAD      MAD  Madrid / Barajas
3113      2013-01-04         MAD      MAD  Madrid / Barajas
3118      2013-01-04         MAD      MAD  Madrid / Barajas
...              ...         ...      ...               ...
20387892  2013-10-11         MAD      MAD  Madrid / Barajas
20387905  2013-10-11         MAD      MAD  Madrid / Barajas
20387947  2013-10-11         MAD      MAD  Madrid / Barajas
20387992  2013-10-11         MAD      MAD  Madrid / Barajas
20388011  2013-10-11         MAD      MAD  Madrid / Barajas
20388224  2013-10-11         MAD      MAD  Madrid / Barajas
20388230  2013-10-11         MAD      MAD  Madrid / Barajas
20388253  2013-10-11         MAD      MAD  Madrid / Barajas
20388305  2013-10-11         MAD      MAD  Madrid / Barajas
20388371  2013-10-12         MAD      MAD  Madrid / Barajas
20388383  2013-10-12         MAD      MAD  Madrid / Barajas
20388536  2013-10-12         MAD      MAD  Madrid / Barajas
20388553  2013-10-12         MAD      MAD  Madrid / Barajas
20388579  2013-10-12         MAD      MAD  Madrid / Barajas
20388662  2013-10-12         MAD      MAD  Madrid / Barajas
20388758  2013-10-12         MAD      MAD  Madrid / Barajas
20388921  2013-10-12         MAD      MAD  Madrid / Barajas
20389019  2013-10-12         MAD      MAD  Madrid / Barajas
20389080  2013-10-12         MAD      MAD  Madrid / Barajas
20389086  2013-10-12         MAD      MAD  Madrid / Barajas
20389089  2013-10-12         MAD      MAD  Madrid / Barajas
20389101  2013-10-12         MAD      MAD  Madrid / Barajas
20389160  2013-10-12         MAD      MAD  Madrid / Barajas
20389206  2013-10-12         MAD      MAD  Madrid / Barajas
20389555  2013-10-13         MAD      MAD  Madrid / Barajas
20389748  2013-10-13         MAD      MAD  Madrid / Barajas
20389834  2013-10-13         MAD      MAD  Madrid / Barajas
20390045  2013-10-13         MAD      MAD  Madrid / Barajas
20390160  2013-10-13         MAD      MAD  Madrid / Barajas
20390196  2013-10-13         MAD      MAD  Madrid / Barajas

[269143 rows x 4 columns]

In [40]:
df["Date"] = pd.to_datetime(df["Date"])

/home/glin/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [ ]:
df = df.copy(deep=True)